libs/shared/contentful/src/lib/queries/eligibility-content-by-plan-ids.ts

```diff
@@ -0,0 +1,117 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Expose, plainToClass, Transform, Type } from 'class-transformer';
+import { IsArray, IsString, MaxLength, ValidateNested } from 'class-validator';
+
+import { graphql } from '../../__generated__/gql';
+
+export const eligibilityContentByPlanIdsQuery = graphql(`
+  query EligibilityContentByPlanIds(
+    $skip: Int!
+    $limit: Int!
+    $locale: String!
+    $stripePlanIds: [String]!
+  ) {
+    purchaseCollection(
+      skip: $skip
+      limit: $limit
+      locale: $locale
+      where: { stripePlanChoices_contains_some: $stripePlanIds }
+    ) {
+      items {
+        stripePlanChoices
+        offering {
+          stripeProductId
+          countries
+          linkedFrom {
+            subGroupCollection(skip: $skip, limit: $limit) {
+              items {
+                groupName
+                offeringCollection(skip: $skip, limit: $limit) {
+                  items {
+                    stripeProductId
+                    countries
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`);
+
+/**
+ * Transform for the `purchaseCollection` field in the Contentful GraphQL response.
+ */
+function parseEligibilityPurchaseResult({ value }: any) {
```

Comment:
While I'd love to not use `any` in any circumstances, these things are the deeply nested ad-hoc generated types which provide no path for getting their types out.

Reply:
Fixed to have its appropriate type.